### PR TITLE
Add RAS comms template service to pipeline

### PR DESCRIPTION
**Pull request by MIslam92**
# Motivation and Context
Automate the deployment of ONSdigital/rm-comms-template-service and to integrate it as part of our pipeline and include it in our other spaces as well.

# What has changed
Mainly the addition of the rm-comms-template-service has been added to the `deployment.yml`, currently added it to all spaces i.e. latest, ci, preprod, prod, 

# How to test?
Check all the passed variables and parameters are the correct ones for all the relevant spaces.
Check that it is part of the main pipeline when it has been flown i.e. https://concourse.onsdigital.uk/teams/rmras/pipelines/sams-amazing-pipeline

# Links
Trello Card: https://trello.com/c/kpHkiXrv/404-add-rm-comms-template-service-to-ras-deploy-m

### CODE REVIEW

File: pipelines/deployment.yml [hunk 1]
```diff
@@ -1359,6 +1392,36 @@ jobs:
...
 
+- name: rm-comms-template-service-ci-deploy
+  serial_groups: [rm-comms-template-service-ci-deploy]
+  plan:
+  - get: rm-comms-template-service-source
+    trigger: true
+  - get: ci-teardown-timer
+    trigger: true
+    passed: [ci-teardown]
+  - put: create-redis
```
Comment (tej99) [inline pipelines/deployment.yml]: Does it need this redis resource?

File: pipelines/deployment.yml [hunk 2]
```diff
@@ -1164,7 +1197,7 @@ jobs:
           repository: paasmule/curl-ssl-git
           tag: "latest"
       inputs:
-        - name: django-oauth2-test-source
+        e- name: django-oauth2-test-source
```
Comment (tej99) [inline pipelines/deployment.yml]: e?

File: pipelines/deployment.yml [hunk 3]
```diff
@@ -1359,6 +1392,36 @@ jobs:
...
+  serial_groups: [rm-comms-template-service-ci-deploy]
+  plan:
+  - get: rm-comms-template-service-source
+    trigger: true
+  - get: ci-teardown-timer
+    trigger: true
+    passed: [ci-teardown]
+  - put: create-redis
+    resource: cf-cli-resource-ci
+    params:
```
Comment (tej99) [inline pipelines/deployment.yml]: Might need to add database resource

File: pipelines/deployment.yml [hunk 4]
```diff
@@ -1359,6 +1392,36 @@ jobs:
...
+      timeout: 1800
+      wait_for_service: true
+  - put: cf-resource-ci
+    params:
+      current_app_name: rm-comms-template-service
+      manifest: rm-comms-template-service/manifest.yml
+      path: rm-comms-template-service-source
+      environment_variables:
+        <<: *ci_security_user
+        <<: *ci_service_endpoints_for_python
```
Comment (tej99) [inline pipelines/deployment.yml]: probably don't need 
```
<<: *ci_service_endpoints_for_python
JSON_SECRET_KEYS: ((ci_json_secret_keys))
        JWT_SECRET: ((ci_jwt_secret))
        SECRET_KEY: ((ci_enrolement_code_encryption_key))
        endpoints_enabled: 'false'
```

File: pipelines/deployment.yml [hunk 5]
```diff
@@ -6,6 +6,8 @@ templates:
       actionSvc_connectionConfig_username: ((ci_security_user_name))
       caseSvc_connectionConfig_username: ((ci_security_user_name))
       caseSvc_connectionConfig_password: ((ci_security_user_password))
+      commsTemplateSvc_connectionConfig_username: ((ci_security_user_name))
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_username

File: pipelines/deployment.yml [hunk 6]
```diff
@@ -193,6 +200,8 @@ templates:
       actionSvc_connectionConfig_username: ((latest_security_user_name))
       caseSvc_connectionConfig_username: ((latest_security_user_name))
       caseSvc_connectionConfig_password: ((latest_security_user_password))
+      commsTemplateSvc_connectionConfig_username: ((latest_security_user_name))
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_username

File: pipelines/deployment.yml [hunk 7]
```diff
@@ -6,6 +6,8 @@ templates:
       actionSvc_connectionConfig_username: ((ci_security_user_name))
       caseSvc_connectionConfig_username: ((ci_security_user_name))
       caseSvc_connectionConfig_password: ((ci_security_user_password))
+      commsTemplateSvc_connectionConfig_username: ((ci_security_user_name))
+      commsTemplateSvc_connectionConfig_password: ((ci_security_user_password))
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_password

File: pipelines/deployment.yml [hunk 8]
```diff
@@ -22,6 +24,8 @@ templates:
       surveySvc_connectionConfig_password: ((ci_security_user_password))
       CASE_USERNAME: ((ci_security_user_name))
       CASE_PASSWORD: ((ci_security_user_password))
+      COMMS_TEMPLATE_USERNAME: ((ci_security_user_name))
```
Comment (tej99) [inline pipelines/deployment.yml]: Probably don't need this no services use it (and next line)

File: pipelines/deployment.yml [hunk 9]
```diff
@@ -193,6 +200,8 @@ templates:
       actionSvc_connectionConfig_username: ((latest_security_user_name))
       caseSvc_connectionConfig_username: ((latest_security_user_name))
       caseSvc_connectionConfig_password: ((latest_security_user_password))
+      commsTemplateSvc_connectionConfig_username: ((latest_security_user_name))
+      commsTemplateSvc_connectionConfig_password: ((latest_security_user_password
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_password

File: pipelines/deployment.yml [hunk 10]
```diff
@@ -209,6 +218,8 @@ templates:
       surveySvc_connectionConfig_password: ((latest_security_user_password))
       CASE_USERNAME: ((latest_security_user_name))
       CASE_PASSWORD: ((latest_security_user_password))
+      COMMS_TEMPLATE_SERVICE_USERNAME: ((latest_security_user_name))
```
Comment (tej99) [inline pipelines/deployment.yml]: don't need these two lines

File: pipelines/deployment.yml [hunk 11]
```diff
@@ -380,6 +394,8 @@ templates:
       actionSvc_connectionConfig_username: ((preprod_security_user_name))
       caseSvc_connectionConfig_username: ((preprod_security_user_name))
       caseSvc_connectionConfig_password: ((preprod_security_user_password))
+      commsTemplateSvc_connectionConfig_username: ((preprod_security_user_name))
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_username

File: pipelines/deployment.yml [hunk 12]
```diff
@@ -380,6 +394,8 @@ templates:
...
+      commsTemplateSvc_connectionConfig_password: ((preprod_security_user_password))
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_password

File: pipelines/deployment.yml [hunk 13]
```diff
@@ -396,6 +412,8 @@ templates:
       surveySvc_connectionConfig_password: ((preprod_security_user_password))
       CASE_USERNAME: ((preprod_security_user_name))
       CASE_PASSWORD: ((preprod_security_user_password))
+      COMMS_TEMPLATE_USERNAME: ((preprod_security_user_name))
```
Comment (tej99) [inline pipelines/deployment.yml]: don't need these two lines

File: pipelines/deployment.yml [hunk 14]
```diff
@@ -567,6 +588,8 @@ templates:
       actionSvc_connectionConfig_username: ((prod_security_user_name))
       caseSvc_connectionConfig_username: ((prod_security_user_name))
       caseSvc_connectionConfig_password: ((prod_security_user_password))
+      commsTemplateSvc_connectionConfig_username: ((prod_security_user_name))
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_username

File: pipelines/deployment.yml [hunk 15]
```diff
@@ -567,6 +588,8 @@ templates:
...
+      commsTemplateSvc_connectionConfig_password: ((prod_security_user_password))
```
Comment (tej99) [inline pipelines/deployment.yml]: commsTemplateService_connectionConfig_password

File: pipelines/deployment.yml [hunk 16]
```diff
@@ -1491,10 +1554,10 @@ jobs:
           source:
             repository: golang
         inputs:
-          - name: rm-survey-service-source
+        - name: rm-survey-service-source
```
Comment (tej99) [inline pipelines/deployment.yml]: Need to fix indents

File: pipelines/deployment.yml [hunk 17]
```diff
@@ -1766,9 +1832,11 @@ jobs:
   - get: rm-collection-exercise-service-source
     passed: [rm-collection-exercise-ci-deploy]
     trigger: true
+  - get: rm-comms-template-service-source
+    passed: [rm-comms-template-service-ci-deploy]
+    trigger: true
   - get: iac-service-source
     passed: [iac-service-ci-deploy]
-    trigger: true
```
Comment (tej99) [inline pipelines/deployment.yml]: i think we need this

File: pipelines/deployment.yml [hunk 18]
```diff
@@ -2162,6 +2230,32 @@ jobs:
         SCHEDULES_VALIDATION_SCHEDULE_DELAY_MILLI_SECONDS: '1000'
         SCHEDULES_DISTRIBUTION_SCHEDULE_DELAY_MILLI_SECONDS: '1000'
 
+- name: rm-comms-template-service-latest-deploy
+  serial_groups: [rm-comms-template-service-latest-deploy]
+  plan:
+  - get: rm-comms-template-service-source
+    trigger: true
+    passed: [ci-rasrm-acceptance-tests, ci-securemessage-acceptance-tests]
+  - put: create-redis
```
Comment (tej99) [inline pipelines/deployment.yml]: don't need redis, but needs a database resource

File: pipelines/deployment.yml [hunk 19]
```diff
@@ -2162,6 +2230,32 @@ jobs:
...
+      timeout: 1800
+      wait_for_service: true
+  - put: cf-resource-latest
+    params:
+      current_app_name: rm-comms-template-service-latest
+      manifest: rm-comms-template-service-source/manifest.yml
+      path: rm-comms-template-service-source
+      environment_variables:
+        <<: *latest_security_user
+        <<: *latest_service_endpoints_for_python
```
Comment (tej99) [inline pipelines/deployment.yml]: scrap  
```
<<: *latest_service_endpoints_for_python
        REDIS_SERVICE: ras-redis
        endpoints_enabled: 'false'
```

File: pipelines/deployment.yml [hunk 20]
```diff
@@ -2281,7 +2375,7 @@ jobs:
           - name: rm-survey-service-source
         outputs:
           - name: rm-survey-service-build
-            path: rm-survey-service
+          path: rm-survey-service
```
Comment (tej99) [inline pipelines/deployment.yml]: indent fix

File: pipelines/deployment.yml [hunk 21]
```diff
@@ -2317,7 +2411,7 @@ jobs:
         - name: sdc-uaa-source
       outputs:
         - name: sdc-uaa-build
-          path: sdc-uaa-build
+        path: sdc-uaa-build
```
Comment (tej99) [inline pipelines/deployment.yml]: indent fix

File: pipelines/deployment.yml [hunk 22]
```diff
@@ -2880,6 +2979,33 @@ disabled-jobs:
...
         SCHEDULES_DISTRIBUTION_SCHEDULE_DELAY_MILLI_SECONDS: '1000'
 
+- name: rm-comms-template-service-preprod-deploy
+  disable_manual_trigger: true
+  serial_groups: [rm-comms-template-service-preprod-deploy]
+  plan:
+  - get: rm-comms-template-service-source
+    trigger: true
+    passed: [preprod-trigger]
+  - put: create-redis
```
Comment (tej99) [inline pipelines/deployment.yml]: - redis, + database

File: pipelines/deployment.yml [hunk 23]
```diff
@@ -2880,6 +2979,33 @@ disabled-jobs:
...
+      timeout: 1800
+      wait_for_service: true
+  - put: cf-resource-preprod
+    params:
+      current_app_name: rm-comms-template-service-concourse-preprod
+      manifest: rm-comms-template-service-source/manifest.yml
+      path: rm-comms-template-service-source
+      environment_variables:
+        <<: *preprod_security_user
+        <<: *preprod_service_endpoints_for_python
```
Comment (tej99) [inline pipelines/deployment.yml]: scrap 
```
<<: *preprod_service_endpoints_for_python
        UAA_CLIENT_ID: 'rm-comms-template-service'
        UAA_CLIENT_SECRET: ((rm-comms-template-service_client_secret))
```

File: pipelines/deployment.yml [hunk 24]
```diff
@@ -3004,7 +3130,7 @@ disabled-jobs:
           - name: rm-survey-service-source
         outputs:
           - name: rm-survey-service-build
-            path: rm-survey-service
+          path: rm-survey-service
```
Comment (tej99) [inline pipelines/deployment.yml]: IF

File: pipelines/deployment.yml [hunk 25]
```diff
@@ -3226,10 +3355,12 @@ disabled-jobs:
     passed: [preprod-smoke-tests]
   - get: rm-actionexporter-service-source
     passed: [preprod-smoke-tests]
-  - get: rm-case-service-source
+  - get: rm-case-service-source-source
```
Comment (tej99) [inline pipelines/deployment.yml]: remove double -source

File: pipelines/deployment.yml [hunk 26]
```diff
@@ -3409,9 +3540,9 @@ disabled-jobs:
           repository: paasmule/curl-ssl-git
           tag: "latest"
       inputs:
-        - name: django-oauth2-test-source
```
Comment (tej99) [inline pipelines/deployment.yml]: IF

File: pipelines/deployment.yml [hunk 27]
```diff
@@ -3409,9 +3540,9 @@ disabled-jobs:
...
+      - name: django-oauth2-test-source
       outputs:
-        - name: django-oauth2-test-source-git-info
```
Comment (tej99) [inline pipelines/deployment.yml]: IF

File: pipelines/deployment.yml [hunk 28]
```diff
@@ -3593,6 +3724,34 @@ disabled-jobs:
...
         SCHEDULES_DISTRIBUTION_SCHEDULE_DELAY_MILLI_SECONDS: '1000'
 
+- name: rm-comms-template-service-prod-deploy
+  disable_manual_trigger: true
+  serial_groups: [rm-comms-template-service-prod-deploy]
+  plan:
+  - get: rm-comms-template-service-source
+    trigger: true
+    passed: [prod-trigger]
+  - put: create-redis
```
Comment (tej99) [inline pipelines/deployment.yml]: - redis + database

File: pipelines/deployment.yml [hunk 29]
```diff
@@ -3593,6 +3724,34 @@ disabled-jobs:
...
+      service_instance: ras-redis
+      timeout: 1800
+      wait_for_service: true
+  - put: cf-resource-prod
+    params:
+      current_app_name: rm-comms-template-service-concourse-prod
+      manifest: rm-comms-template-service/manifest.yml
+      path: rm-comms-template-service-source
+      environment_variables:
+        <<: *prod_service_endpoints_for_python
```
Comment (tej99) [inline pipelines/deployment.yml]: scrap

```
<<: *prod_service_endpoints_for_python
REDIS_SERVICE: ras-redis
        UAA_CLIENT_ID: 'rm-comms-template-service'
        UAA_CLIENT_SECRET: ((prod_rm_comms_template_service_client_secret)
```

File: pipelines/deployment.yml [hunk 30]
```diff
@@ -1359,6 +1386,31 @@ jobs:
...
+    params:
+      command: create-service
+      service: rds
+      plan: shared-psql
+      service_instance: rm-commstemplate-db
+      timeout: 300
+      wait_for_service: true
+  - put: cf-resource-ci
+    params:
+      current_app_name: rm-comms-template-service
```
Comment (tej99) [inline pipelines/deployment.yml]: rm-comms-template-service-ci?

**Review by tej99 — APPROVED** [review]

**Review by shahi645 — APPROVED** [review]
LGTM. tested with Sam by deploying a test pipeline

**Pull request merged**